### PR TITLE
Make Owners a separate service.

### PR DESCRIPTION
**Pull request by lannka**
- Move Owners to a separate service, with only one method for now
- Any caller of that method should call with service directly instead of via the base-element -> custom-element chain.

Let me know what you think. Will migrate all other methods in this way one after another to make sure every extension is well tested.

for #23311

### CODE REVIEW

File: src/service/resources-impl.js [hunk 1]
```diff
@@ -74,7 +75,7 @@ let ChangeSizeRequestDef;
 /**
  * @interface
  */
-class OwnersDef {
+class OwnersDef extends Owners {
```
Comment (choumx) [inline src/service/resources-impl.js]: I assume this "extends" relationship is temporary, right? `OwnersDef` in this file will just be moved into owners-impl.js? 

A4A then will register and use its own implementation of the interface?

File: src/service/resources-impl.js [hunk 2]
```diff
@@ -2723,5 +2716,24 @@ export let SizeDef;
...
 export function installResourcesServiceForDoc(ampdoc) {
-  registerServiceBuilderForDoc(ampdoc, 'resources', Resources);
+  registerServiceBuilderForDoc(ampdoc, 'resources', getResources);
+}
+
+/**
+ * @param {!./ampdoc-impl.AmpDoc} ampdoc
+ */
+export function installOwnersServiceForDoc(ampdoc) {
+  registerServiceBuilderForDoc(ampdoc, 'owners', getResources);
```
Comment (choumx) [inline src/service/resources-impl.js]: Sorry for missing context here. Why split resources into three services? 

Will non-A4A return the same object instance for the three services (resources, owners, mutator)?

File: src/service/resources-impl.js (hunk 1 above)
Comment (lannka) [reply src/service/resources-impl.js]: Yes. The goal of this PR is layout the structure, so we can move the remaining methods from OwnersDef to Owners in parallel. 

a4a will have its own impls of those interfaces.

File: src/service/resources-impl.js (hunk 2 above)
Comment (lannka) [reply src/service/resources-impl.js]: This is after a discussion with @dvoytenko . It aligns with the goal of the Bento initiative to split this big services into smaller ones. Some of them are core rendering critical (`resources`), some are not (`mutators`). `owners` should finally go away after `layers` landing (@jridgewell) 

As you can see in this PR, we return same instance for those services in regular AMP. But eventually, they should be split up (as part of the Bento project).

Let me know if this all makes sense to you.

**Comment by lannka** [comment]
woops, moving one method failed 156 unit tests 
:-(

File: src/service/resources-impl.js (hunk 2 above)
Comment (choumx) [reply src/service/resources-impl.js]: Sounds good.

Comment (jridgewell) [inline src/service/resources-impl.js]: I don't understand this call. It's returning `resources`?

Comment (lannka) [reply src/service/resources-impl.js]: this line of code had a bug. Fixed.

**Comment by lannka** [comment]
fixed the test failures. It was caused by 
- we installAmpdocServices to both karma window & test iframe window.
- The previous code mixed resources instance of the 2 window 

PTAL

File: src/service/resources-impl.js [hunk 3]
```diff
@@ -2725,3 +2718,12 @@ export let SizeDef;
 export function installResourcesServiceForDoc(ampdoc) {
   registerServiceBuilderForDoc(ampdoc, 'resources', Resources);
 }
+
+/**
+ * @param {!./ampdoc-impl.AmpDoc} ampdoc
+ */
+export function installOwnersServiceForDoc(ampdoc) {
+  registerServiceBuilderForDoc(ampdoc, 'owners', ampdoc => {
+    return Services.resourcesForDoc(ampdoc);
```
Comment (choumx) [inline src/service/resources-impl.js]: Does this rely on Dima's change to make FIE an ampdoc?

File: src/service/owners-impl.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-unused-vars */
+/**
+ * TODO(lannka): migrate all methods from OwnersDef to here.
```
Comment (choumx) [inline src/service/owners-impl.js]: This will happen in a separate PR? This intermediate state is a bit weird.

Comment (lannka) [reply src/service/owners-impl.js]: a couple of PRs

File: src/service/resources-impl.js (hunk 3 above)
Comment (lannka) [reply src/service/resources-impl.js]: no

**Pull request merged**